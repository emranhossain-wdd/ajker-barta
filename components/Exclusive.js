import { formatDistanceToNow } from 'date-fns';
import { useRouter } from 'next/router';
import React, { useState } from 'react'
import styles from '../styles/exclusive.module.css'
const Exclusive = ({ newses }) => {
  const latest = newses?.reverse()
  const router = useRouter()
  const [isLetest, setIsLetest] = useState(true)
  let count = 1
  return (
    <div data-testid='exclusiveId' className='pb-5'>
      <div className="container">
        <div className="grid lg:grid-cols-2 gap-3">
          <div>
            <button onClick={() => setIsLetest(true)} className={isLetest ? 'py-1 slim-border rounded-sm mb-3 px-4 border-blue-400 mr-2 bg-blue-500 text-white' : 'py-1 slim-border rounded-sm mb-3 px-4 bg-white border-blue-400 mr-2'}>Letest</button>
            <button onClick={() => setIsLetest(false)} className={!isLetest ? 'py-1 slim-border rounded-sm mb-3 px-4 border-blue-400 mr-2 bg-blue-500 text-white' : 'py-1 slim-border rounded-sm mb-3 px-4 bg-white border-blue-400 mr-2'}>Trending</button>
            {
              isLetest ? <div className='grid sm:grid-cols-2 gap-3'>
                {
                  latest?.slice(0, 8).map(news => <div onClick={() => router.push(`/${news.category}/${news.subCategory}/${news?._id}`)} className=' relative p-6 shadow-md rounded-r-lg border-r-4 border-red-600 cursor-pointer' key={news._id}>
                    <h1 className='leading-7 relative z-10 text-lg font-medium text-gray-800 hover:text-red-600 transition-colors duration-300 cursor-pointer'>{news?.heading}</h1>
                    <p style={{ transform: "translate(-50%, -50%)", }} className='absolute top-1/2 left-3/4 text-8xl font-bold text-blue-100 '>{count++}</p>
                  </div>)
                }
              </div> : <div className='grid sm:grid-cols-2 gap-3'>
                {
                  latest?.slice(8, 16).map(news => <div onClick={() => router.push(`/news/${news._id}`)} className=' relative p-6 shadow-md rounded-r-lg border-r-4 border-red-600 cursor-pointer' key={news._id}>
                    <h1 className='leading-7 relative z-10 text-lg font-medium text-gray-800 hover:text-red-600 transition-colors duration-300 cursor-pointer'>{news?.heading}</h1>
                    <p style={{ transform: "translate(-50%, -50%)", }} className='absolute top-1/2 left-3/4 text-8xl font-bold text-blue-100 '>{count++}</p>
                  </div>)
                }
              </div>
            }

          </div>
          <div>
            <h1 className={styles.exclusive}>Satireday</h1>
            <div className='grid sm:grid-cols-2 gap-3'>
              {
                latest?.slice(16, 20).map(news => <div className='cursor-pointer imageEffect' onClick={() => router.push(`/${news.category}/${news.subCategory}/${news?._id}`)} key={news?._id}>
                  <div className='overflow-hidden'>
                    <img className='w-full h-32 object-cover' src={news?.images.img1} alt="" />
                  </div>
                  <h1 className='text-lg my-1 font-medium leading-5 hover:text-red-600 transition-colors duration-300 cursor-pointer'>{news?.heading}</h1>
                  <p className='text-sm leading-4'>{news?.description[0]?.slice(0, 100)}</p>
                  <p className="px-2 mt-2 py-1 font-medium text-sm rounded-full bg-gray-100 w-fit text-blue-500">{`${formatDistanceToNow(new Date(news.publishedDate))} ago`}</p>
                </div>)
              }
            </div>
          </div>
        </div>
      </div>
    </div>
  );
};

export default Exclusive;