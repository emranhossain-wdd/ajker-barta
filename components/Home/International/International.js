import React from 'react';
// import styles from './International.module.css';

const newsData = [
    {
        "author": "Dan Peck",
        "title": "Hurricane Hanna makes landfall around 5 p.m. on Saturday.",
        "description": "Hurricane Hanna battered southern Texas with sustained winds of 75 mph and continued to deliver heavy rain and flash flooding as it moved inland late Saturday.",
        "url": "https://abcnews.go.com/US/hurricane-hanna-makes-landfall-texas/story?id=71985566",
        "source": "ABC News",
        "image": "https://s.abcnews.com/images/US/hanna-swimmer-mo_hpMain_20200725-163152_2_4x3t_384.jpg",
        "category": "general",
        "language": "en",
        "country": "us",
        "published_at": "2020-07-26T01:04:23+00:00"
    },
    {
        "author": "TMZ Staff",
        "title": "Nicki Minaj's Husband Gets Permission To Be There For Baby's Birth",
        "description": "Kenneth can be in the room when Nicki gives birth ... a judge just granted his request to tweak his pre-trial release conditions so he can travel with Nicki. With the court's order in place, KP can travel with Nicki periodically on biz…",
        "url": "https://www.tmz.com/2020/07/30/nicki-minaj-husband-asks-judge-be-there-child-birth/",
        "source": "TMZ.com",
        "image": "https://imagez.tmz.com/image/c1/4by3/2020/07/30/c115ad2dc849438a97a0ad3097b416df_md.jpg",
        "category": "general",
        "language": "en",
        "country": "us",
        "published_at": "2020-08-01T05:34:47+00:00"
    },
    {
        "author": "Michael Rothstein",
        "title": "New Lions safety Jayron Kearse suspended three games",
        "description": "Kearse, 26, signed with the Lions in March after four seasons in Minnesota, where he played in 62 games with five starts, making 79 tackles and defending eight passes.",
        "url": "https://www.espn.com/nfl/story/_/id/29572640/new-lions-safety-jayron-kearse-suspended-three-games",
        "source": "ESPN",
        "image": "https://a4.espncdn.com/combiner/i?img=%2Fphoto%2F2020%2F0111%2Fr651071_1296x729_16%2D9.jpg",
        "category": "sports",
        "language": "en",
        "country": "us",
        "published_at": "2020-07-31T23:23:14+00:00"
    },
    {
        "author": "Dan Peck",
        "title": "Hurricane Hanna makes landfall around 5 p.m. on Saturday.",
        "description": "Hurricane Hanna battered southern Texas with sustained winds of 75 mph and continued to deliver heavy rain and flash flooding as it moved inland late Saturday.",
        "url": "https://abcnews.go.com/US/hurricane-hanna-makes-landfall-texas/story?id=71985566",
        "source": "ABC News",
        "image": "https://s.abcnews.com/images/US/hanna-swimmer-mo_hpMain_20200725-163152_2_4x3t_384.jpg",
        "category": "general",
        "language": "en",
        "country": "us",
        "published_at": "2020-07-26T01:04:23+00:00"
    },
    {
        "author": "TMZ Staff",
        "title": "Nicki Minaj's Husband Gets Permission To Be There For Baby's Birth",
        "description": "Kenneth can be in the room when Nicki gives birth ... a judge just granted his request to tweak his pre-trial release conditions so he can travel with Nicki. With the court's order in place, KP can travel with Nicki periodically on biz…",
        "url": "https://www.tmz.com/2020/07/30/nicki-minaj-husband-asks-judge-be-there-child-birth/",
        "source": "TMZ.com",
        "image": "https://imagez.tmz.com/image/c1/4by3/2020/07/30/c115ad2dc849438a97a0ad3097b416df_md.jpg",
        "category": "general",
        "language": "en",
        "country": "us",
        "published_at": "2020-08-01T05:34:47+00:00"
    },
    {
        "author": "Michael Rothstein",
        "title": "New Lions safety Jayron Kearse suspended three games",
        "description": "Kearse, 26, signed with the Lions in March after four seasons in Minnesota, where he played in 62 games with five starts, making 79 tackles and defending eight passes.",
        "url": "https://www.espn.com/nfl/story/_/id/29572640/new-lions-safety-jayron-kearse-suspended-three-games",
        "source": "ESPN",
        "image": "https://a4.espncdn.com/combiner/i?img=%2Fphoto%2F2020%2F0111%2Fr651071_1296x729_16%2D9.jpg",
        "category": "sports",
        "language": "en",
        "country": "us",
        "published_at": "2020-07-31T23:23:14+00:00"
    },
    {
        "author": "Dan Peck",
        "title": "Hurricane Hanna makes landfall around 5 p.m. on Saturday.",
        "description": "Hurricane Hanna battered southern Texas with sustained winds of 75 mph and continued to deliver heavy rain and flash flooding as it moved inland late Saturday.",
        "url": "https://abcnews.go.com/US/hurricane-hanna-makes-landfall-texas/story?id=71985566",
        "source": "ABC News",
        "image": "https://s.abcnews.com/images/US/hanna-swimmer-mo_hpMain_20200725-163152_2_4x3t_384.jpg",
        "category": "general",
        "language": "en",
        "country": "us",
        "published_at": "2020-07-26T01:04:23+00:00"
    },
    {
        "author": "TMZ Staff",
        "title": "Nicki Minaj's Husband Gets Permission To Be There For Baby's Birth",
        "description": "Kenneth can be in the room when Nicki gives birth ... a judge just granted his request to tweak his pre-trial release conditions so he can travel with Nicki. With the court's order in place, KP can travel with Nicki periodically on biz…",
        "url": "https://www.tmz.com/2020/07/30/nicki-minaj-husband-asks-judge-be-there-child-birth/",
        "source": "TMZ.com",
        "image": "https://imagez.tmz.com/image/c1/4by3/2020/07/30/c115ad2dc849438a97a0ad3097b416df_md.jpg",
        "category": "general",
        "language": "en",
        "country": "us",
        "published_at": "2020-08-01T05:34:47+00:00"
    },
    {
        "author": "Michael Rothstein",
        "title": "New Lions safety Jayron Kearse suspended three games",
        "description": "Kearse, 26, signed with the Lions in March after four seasons in Minnesota, where he played in 62 games with five starts, making 79 tackles and defending eight passes.",
        "url": "https://www.espn.com/nfl/story/_/id/29572640/new-lions-safety-jayron-kearse-suspended-three-games",
        "source": "ESPN",
        "image": "https://a4.espncdn.com/combiner/i?img=%2Fphoto%2F2020%2F0111%2Fr651071_1296x729_16%2D9.jpg",
        "category": "sports",
        "language": "en",
        "country": "us",
        "published_at": "2020-07-31T23:23:14+00:00"
    },
]

const International = () => {
    return (
        <div className='container mx-auto'>
            <div className="grid grid-cols-12 gap-2">
                <div className='md:col-span-8 sm:col-span-6 ' style={{ height: '100%' }}>
                    {/* <div className={`${styles.newCard} " border border-gray-400 p-2"`}>
                        <a href="#" className={styles.newsCard__cardLink}></a>
                        <img src="https://a4.espncdn.com/combiner/i?img=%2Fphoto%2F2020%2F0111%2Fr651071_1296x729_16%2D9.jpg" className={styles.newsCard__image} />
                        <div className={styles.newsCard__textWrapper}>
                            <h2 className={styles.newsCard__title}>Amazing First Title</h2>
                            <div className={styles.newsCard__postDate}>Jan 29, 2018</div>
                            <div className={styles.newsCard__detailsWrapper}>
                                <p className={styles.newsCard__excerpt}>Lorem ipsum dolor sit amet consectetur adipisicing elit. Est pariatur nemo tempore repellat? Ullam sed officia iure architecto deserunt distinctio, pariatur&hellip;</p>
                                <a href="#" className={styles.newsCard__readMore}>Read more <i className="fas fa-long-arrow-alt-right"></i></a>
                            </div>
                        </div>
                    </div> */}
                    <div className="news-card" >
                        <a href="#" className="news-card__card-link"></a>
                        <img src="https://images.pexels.com/photos/127513/pexels-photo-127513.jpeg?auto=compress&cs=tinysrgb&h=750&w=1260" alt="" className="news-card__image" />
                        <div className="news-card__text-wrapper">
                            <h2 className="news-card__title">Amazing First Title</h2>
                            <div className="news-card__post-date">Jan 29, 2018</div>
                            <div className="news-card__details-wrapper">
                                <p className="news-card__excerpt">Lorem ipsum dolor sit amet consectetur adipisicing elit. Est pariatur nemo tempore repellat? Ullam sed officia iure architecto deserunt distinctio, pariatur&hellip;</p>
                                <a href="#" className="news-card__read-more">Read more <i className="fas fa-long-arrow-alt-right"></i></a>
                            </div>
                        </div>
                    </div>
                </div>


                <div className="md:col-span-4 sm:col-span-6">
                    <div className='grid grid-cols-1 gap-1'>
                        {/* <div className={`${styles.newCard} " border border-gray-400 p-2"`}>
                        <a href="#" className={styles.newsCard__cardLink}></a>
                        <img src="https://a4.espncdn.com/combiner/i?img=%2Fphoto%2F2020%2F0111%2Fr651071_1296x729_16%2D9.jpg" className={styles.newsCard__image} />
                        <div className={styles.newsCard__textWrapper}>
                            <h2 className={styles.newsCard__title}>Amazing First Title</h2>
                            <div className={styles.newsCard__postDate}>Jan 29, 2018</div>
                            <div className={styles.newsCard__detailsWrapper}>
                                <p className={styles.newsCard__excerpt}>Lorem ipsum dolor sit amet consectetur adipisicing elit. Est pariatur nemo tempore repellat? Ullam sed officia iure architecto deserunt distinctio, pariatur&hellip;</p>
                                <a href="#" className={styles.newsCard__readMore}>Read more <i className="fas fa-long-arrow-alt-right"></i></a>
                            </div>
                        </div>
                    </div>
                    <div className={`${styles.newCard} " border border-gray-400 p-2"`}>
                        <a href="#" className={styles.newsCard__cardLink}></a>
                        <img src="https://a4.espncdn.com/combiner/i?img=%2Fphoto%2F2020%2F0111%2Fr651071_1296x729_16%2D9.jpg" className={styles.newsCard__image} />
                        <div className={styles.newsCard__textWrapper}>
                            <h2 className={styles.newsCard__title}>Amazing First Title</h2>
                            <div className={styles.newsCard__postDate}>Jan 29, 2018</div>
                            <div className={styles.newsCard__detailsWrapper}>
                                <p className={styles.newsCard__excerpt}>Lorem ipsum dolor sit amet consectetur adipisicing elit. Est pariatur nemo tempore repellat? Ullam sed officia iure architecto deserunt distinctio, pariatur&hellip;</p>
                                <a href="#" className={styles.newsCard__readMore}>Read more <i className="fas fa-long-arrow-alt-right"></i></a>
                            </div>
                        </div>
                    </div> */}
                        <div>
                            <div className="news-card">
                                <a href="#" className="news-card__card-link"></a>
                                <img src="https://images.pexels.com/photos/127513/pexels-photo-127513.jpeg?auto=compress&cs=tinysrgb&h=750&w=1260" alt="" className="news-card__image" />
                                <div className="news-card__text-wrapper">
                                    <h2 className="news-card__title">Amazing First Title</h2>
                                    <div className="news-card__post-date">Jan 29, 2018</div>
                                    <div className="news-card__details-wrapper">
                                        <p className="news-card__excerpt">Lorem ipsum dolor sit amet consectetur adipisicing elit. </p>
                                        <a href="#" className="news-card__read-more">Read more <i className="fas fa-long-arrow-alt-right"></i></a>
                                    </div>
                                </div>
                            </div>
                        </div>
                        <div>
                            <div className="news-card">
                                <a href="#" className="news-card__card-link"></a>
                                <img src="https://images.pexels.com/photos/127513/pexels-photo-127513.jpeg?auto=compress&cs=tinysrgb&h=750&w=1260" alt="" className="news-card__image" />
                                <div className="news-card__text-wrapper">
                                    <h2 className="news-card__title">Amazing First Title</h2>
                                    <div className="news-card__post-date">Jan 29, 2018</div>
                                    <div className="news-card__details-wrapper">
                                        <p className="news-card__excerpt">Lorem ipsum dolor sit amet consectetur adipisicing elit. </p>
                                        <a href="#" className="news-card__read-more">Read more <i className="fas fa-long-arrow-alt-right"></i></a>
                                    </div>
                                </div>
                            </div>
                        </div>
                    </div>
                </div>

            </div>
        </div>
    );
};

export default International;