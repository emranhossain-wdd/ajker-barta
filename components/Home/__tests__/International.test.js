import React from 'react'
import ReactDOM from 'react-dom'
import renderer from 'react-test-renderer'
import International from '../International/International'
import { render } from '@testing-library/react'
import { testArr } from '../../__tests__/Exclusive.test'

it('render International component without crashing', () => {
  const div = document.createElement('div')
  ReactDOM.render(<International />, div)
})

it('testing International with props', () => {
  const { getByTestId } = render(<International testArr={testArr} />)
  expect(getByTestId('internationalId')).toBeTruthy()
})
it('maching snapshot', () => {
  const tree = renderer.create(<International />)
  expect(tree).toMatchSnapshot()
})