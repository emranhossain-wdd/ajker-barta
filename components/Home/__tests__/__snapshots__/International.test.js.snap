// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`maching snapshot 1`] = `
<div
  className="pb-5"
  data-testid="exclusiveId"
>
  <div
    className="container"
  >
    <div
      className="grid lg:grid-cols-2 gap-3"
    >
      <div>
        <button
          className="py-1 slim-border rounded-sm mb-3 px-4 border-blue-400 mr-2 bg-blue-500 text-white"
          onClick={[Function]}
        >
          Letest
        </button>
        <button
          className="py-1 slim-border rounded-sm mb-3 px-4 bg-white border-blue-400 mr-2"
          onClick={[Function]}
        >
          Trending
        </button>
        <div
          className="grid sm:grid-cols-2 gap-3"
        />
      </div>
      <div>
        <h1
          className="exclusive"
        >
          Satireday
        </h1>
        <div
          className="grid sm:grid-cols-2 gap-3"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`maching snapshot 2`] = `
<div
  className="container"
  data-testid="internationalId"
>
  <h2
    className="text-xl font-medium mt-5 mb-2"
  >
    <span
      className="text-blue-900"
    >
      <a
        href="/international"
        onClick={[Function]}
        onMouseEnter={[Function]}
      >
        International
      </a>
       
    </span>
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium text-red-600 css-i4bv87-MuiSvgIcon-root"
      data-testid="ChevronRightIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M10 6 8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
      />
    </svg>
  </h2>
  <div
    className="grid grid-cols-12 gap-4"
  >
    <div
      className="col-span-12 md:col-span-6 mb-3 lg:mb-0"
    >
      <div
        className="p-3 rounded-md border-r border-b"
      />
    </div>
    <div
      className="col-span-12 md:col-span-6"
    >
      <div
        className=""
      />
    </div>
  </div>
</div>
`;
